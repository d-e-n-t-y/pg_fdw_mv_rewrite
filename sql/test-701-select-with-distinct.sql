-- Anticipate message "query requires unsupported features"
set mv_rewrite.log_match_progress = 'true';

set mv_rewrite.log_match_progress = 'true';
explain (VERBOSE, COSTS OFF)
select distinct key, hidden, count (value)
  from test
  group by key, hidden;

select distinct key, hidden, count (value)
  from test
  group by key, hidden;
