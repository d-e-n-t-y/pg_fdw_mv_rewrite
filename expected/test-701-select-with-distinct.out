-- Anticipate message "query requires unsupported features"
-- We don't actually care about the plan itself, only the log output.
set mv_rewrite.log_match_progress = 'true';
set mv_rewrite.rewrite_enabled_for_tables = '';
-- Anticipate no rewrite
DO LANGUAGE plpgsql $$ DECLARE t text; BEGIN FOR t IN
	explain (VERBOSE, COSTS OFF)
		select distinct key, hidden, count (value)
  		from test
  		group by key, hidden
LOOP IF t LIKE '%Rewritten%' THEN RAISE 'Rewritten'; END IF; END LOOP; END; $$;
INFO:  mv_rewrite_add_rewritten_mv_paths: MV rewrite not enabled for one or more table in the query.
INFO:  mv_rewrite_add_rewritten_mv_paths: MV rewrite not enabled for one or more table in the query.
select distinct key, hidden, count (value)
  from test
  group by key, hidden;
INFO:  mv_rewrite_add_rewritten_mv_paths: MV rewrite not enabled for one or more table in the query.
INFO:  mv_rewrite_add_rewritten_mv_paths: MV rewrite not enabled for one or more table in the query.
 key  | hidden  |  count  
------+---------+---------
 key1 | hidden1 |      10
 key2 | hidden2 | 1000000
(2 rows)

